{"uid":"ba3be7a8f03e4a45","name":"Тест удаления из избранного без аутентификации","fullName":"tests.user.test_user_api.TestUserAPI#test_remove_from_favourites_unauthorized","historyId":"1033939dd05afa312f53a1fdaa5c410d","time":{"start":1746398074195,"stop":1746398074219,"duration":24},"description":"Проверяем, что неавторизованный пользователь получает ошибку 401.","descriptionHtml":"<p>Проверяем, что неавторизованный пользователь получает ошибку 401.</p>\n","status":"failed","statusMessage":"AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nBody of the answer:\n{\"message\":\"No token provided.\"}","statusTrace":"self = <tests.user.test_user_api.TestUserAPI object at 0x7fa44f88e780>\nuser_client = <api.user.client.UserClient object at 0x7fa44e2507d0>\n\n    @allure.feature(\"Избранное пользователя (DELETE /api/user/favourites/{id})\")\n    @allure.story(\"Удаление из избранного\")\n    @allure.title(\"Тест удаления из избранного без аутентификации\")\n    @allure.description(\"Проверяем, что неавторизованный пользователь получает ошибку 401.\")\n    @allure.severity(allure.severity_level.CRITICAL)\n    @pytest.mark.negative\n    def test_remove_from_favourites_unauthorized(self, user_client: UserClient) -> None:\n        \"\"\"\n        Проверка удаления из избранного без аутентификации.\n    \n        Ожидаемый результат: статус 401 Unauthorized.\n        \"\"\"\n        logger.info(\"Тест: Удаление из избранного без авторизации (DELETE ...)\")\n>       response = user_client.remove_from_favourites(request_id=\"any-id\", expected_status=401)  # type: ignore\n\ntests/user/test_user_api.py:136: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \napi/user/client.py:82: in remove_from_favourites\n    processed_response = self._handle_response(response, expected_status)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.user.client.UserClient object at 0x7fa44e2507d0>\nresponse = <APIResponse url='http://localhost:4040/api/user/favourites/any-id' status=403 status_text='Forbidden'>\nexpected_status = 401, response_model = None\n\n    def _handle_response(\n        self,\n        response: APIResponse,\n        expected_status: int,\n        response_model: type[T] | None = None,\n    ) -> T | APIResponse:\n        \"\"\"\n        A generic method to handle the API response.\n    \n        Checks the status code and, if a model is specified, validates the response body against it.\n    \n        Args:\n            response: APIResponse object received from HTTPClient.\n            expected_status: Expected HTTP status code.\n            response_model: Optional Pydantic model class for validating the response body.\n    \n        Returns:\n            An instance of response_model if the validation was successful.\n            Raw APIResponse object if response_model is not specified or validation is not required.\n    \n        Raises:\n            AssertionError: If the actual status of the code does not match the expected_status,\n                           or if it failed to parse JSON,\n                           or if the Pydantic model validation failed.\n        \"\"\"\n        self.logger.debug(\n            \"Handling response: Status=%s, Expected=%s, URL=%s\",\n            response.status,\n            expected_status,\n            response.url,\n        )\n    \n        allure.attach(\n            name=f\"Status response code: {response.status} (Expected: {expected_status})\",\n            body=str(response.status),\n            attachment_type=allure.attachment_type.TEXT,\n        )\n    \n>       assert response.status == expected_status, (\n            f\"The status was pending {expected_status}, but received {response.status}. \"\n            f\"URL: {response.url}\\nBody of the answer:\\n{response.text()}\"\n        )\nE       AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nE       Body of the answer:\nE       {\"message\":\"No token provided.\"}\n\napi/base_api.py:69: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"pytestconfig","time":{"start":1746398072889,"stop":1746398072889,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"delete_output_dir","time":{"start":1746398072889,"stop":1746398072889,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"_verify_url","time":{"start":1746398072890,"stop":1746398072890,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_instance","time":{"start":1746398072890,"stop":1746398073189,"duration":299},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"base_url","time":{"start":1746398072890,"stop":1746398072890,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"api_request_context","time":{"start":1746398073189,"stop":1746398073210,"duration":21},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"http_client","time":{"start":1746398073210,"stop":1746398073210,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"user_client","time":{"start":1746398074095,"stop":1746398074096,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Проверяем, что неавторизованный пользователь получает ошибку 401.","status":"failed","statusMessage":"AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nBody of the answer:\n{\"message\":\"No token provided.\"}","statusTrace":"self = <tests.user.test_user_api.TestUserAPI object at 0x7fa44f88e780>\nuser_client = <api.user.client.UserClient object at 0x7fa44e2507d0>\n\n    @allure.feature(\"Избранное пользователя (DELETE /api/user/favourites/{id})\")\n    @allure.story(\"Удаление из избранного\")\n    @allure.title(\"Тест удаления из избранного без аутентификации\")\n    @allure.description(\"Проверяем, что неавторизованный пользователь получает ошибку 401.\")\n    @allure.severity(allure.severity_level.CRITICAL)\n    @pytest.mark.negative\n    def test_remove_from_favourites_unauthorized(self, user_client: UserClient) -> None:\n        \"\"\"\n        Проверка удаления из избранного без аутентификации.\n    \n        Ожидаемый результат: статус 401 Unauthorized.\n        \"\"\"\n        logger.info(\"Тест: Удаление из избранного без авторизации (DELETE ...)\")\n>       response = user_client.remove_from_favourites(request_id=\"any-id\", expected_status=401)  # type: ignore\n\ntests/user/test_user_api.py:136: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \napi/user/client.py:82: in remove_from_favourites\n    processed_response = self._handle_response(response, expected_status)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.user.client.UserClient object at 0x7fa44e2507d0>\nresponse = <APIResponse url='http://localhost:4040/api/user/favourites/any-id' status=403 status_text='Forbidden'>\nexpected_status = 401, response_model = None\n\n    def _handle_response(\n        self,\n        response: APIResponse,\n        expected_status: int,\n        response_model: type[T] | None = None,\n    ) -> T | APIResponse:\n        \"\"\"\n        A generic method to handle the API response.\n    \n        Checks the status code and, if a model is specified, validates the response body against it.\n    \n        Args:\n            response: APIResponse object received from HTTPClient.\n            expected_status: Expected HTTP status code.\n            response_model: Optional Pydantic model class for validating the response body.\n    \n        Returns:\n            An instance of response_model if the validation was successful.\n            Raw APIResponse object if response_model is not specified or validation is not required.\n    \n        Raises:\n            AssertionError: If the actual status of the code does not match the expected_status,\n                           or if it failed to parse JSON,\n                           or if the Pydantic model validation failed.\n        \"\"\"\n        self.logger.debug(\n            \"Handling response: Status=%s, Expected=%s, URL=%s\",\n            response.status,\n            expected_status,\n            response.url,\n        )\n    \n        allure.attach(\n            name=f\"Status response code: {response.status} (Expected: {expected_status})\",\n            body=str(response.status),\n            attachment_type=allure.attachment_type.TEXT,\n        )\n    \n>       assert response.status == expected_status, (\n            f\"The status was pending {expected_status}, but received {response.status}. \"\n            f\"URL: {response.url}\\nBody of the answer:\\n{response.text()}\"\n        )\nE       AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nE       Body of the answer:\nE       {\"message\":\"No token provided.\"}\n\napi/base_api.py:69: AssertionError","steps":[{"name":"Удаление запроса из избранного","time":{"start":1746398074195,"stop":1746398074218,"duration":23},"status":"failed","statusMessage":"AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nBody of the answer:\n{\"message\":\"No token provided.\"}\n","statusTrace":"  File \"/app/.venv/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/app/api/user/client.py\", line 82, in remove_from_favourites\n    processed_response = self._handle_response(response, expected_status)\n  File \"/app/api/base_api.py\", line 69, in _handle_response\n    assert response.status == expected_status, (\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[],"attachments":[{"uid":"36e69e1a55b4f968","name":"Status Code: 403","source":"36e69e1a55b4f968.txt","type":"text/plain","size":3},{"uid":"8789c84940b76fc8","name":"Request URL","source":"8789c84940b76fc8.uri","type":"text/uri-list","size":48},{"uid":"fd2e53c95aeadcf4","name":"Response Headers (JSON)","source":"fd2e53c95aeadcf4.json","type":"application/json","size":316},{"uid":"e43dce440bef0d55","name":"Response Body (JSON)","source":"e43dce440bef0d55.json","type":"application/json","size":39},{"uid":"3290ff137e46ae86","name":"Status response code: 403 (Expected: 401)","source":"3290ff137e46ae86.txt","type":"text/plain","size":3}],"parameters":[{"name":"request_id","value":"'any-id'"},{"name":"expected_status","value":"401"}],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":5,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"d3c9471eb1de1998","name":"log","source":"d3c9471eb1de1998.txt","type":"text/plain","size":454},{"uid":"d1da774e82abf2c1","name":"stderr","source":"d1da774e82abf2c1.txt","type":"text/plain","size":539}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":1,"attachmentsCount":7,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"api_request_context::0","time":{"start":1746398074447,"stop":1746398074453,"duration":6},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_instance::0","time":{"start":1746398074454,"stop":1746398074462,"duration":8},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"epic","value":"Управление пользователем"},{"name":"severity","value":"critical"},{"name":"feature","value":"Избранное пользователя (DELETE /api/user/favourites/{id})"},{"name":"story","value":"Удаление из избранного"},{"name":"tag","value":"negative"},{"name":"tag","value":"user"},{"name":"parentSuite","value":"tests.user"},{"name":"suite","value":"test_user_api"},{"name":"subSuite","value":"TestUserAPI"},{"name":"host","value":"fv-az2031-666"},{"name":"thread","value":"1-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.user.test_user_api"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":3,"broken":0,"skipped":0,"passed":0,"unknown":0,"total":3},"items":[{"uid":"4cc2350ed69f6484","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/91//#testresult/4cc2350ed69f6484","status":"failed","statusDetails":"AssertionError: The status was pending 401, but received 403. URL: http://localhost:4040/api/user/favourites/any-id\nBody of the answer:\n{\"message\":\"No token provided.\"}","time":{"start":1746395043626,"stop":1746395043677,"duration":51}},{"uid":"b4e53dde74e951e4","reportUrl":"https://ZhikharevAl.github.io/fake_it_till_you_make_it/90//#testresult/b4e53dde74e951e4","status":"failed","statusDetails":"AssertionError: The status was pending 401, but received 500. URL: http://localhost:4040/api/user/favourites/any-id\nBody of the answer:\nPlanned Server Error","time":{"start":1746394513569,"stop":1746394513638,"duration":69}}]},"tags":["negative","user"]},"source":"ba3be7a8f03e4a45.json","parameterValues":[]}